### PR TITLE
Feature/react fc conversion

### PR DESCRIPTION
**Pull request by emclaug2**
Changes proposed in this Pull Request:
- Add prettier and linting to react storybook
- Convert existing react components to functional components
- Convert existing unit tests to typescript
- Update DrawerLayout API with feedback from AO

### CODE REVIEW

**Comment by emclaug2** [comment]
I did not convert the Showcase demo over to TS because it might be a good idea to keep it as a JS project so we can test out component integration into a JS project.

**Comment by emclaug2** [comment]
Barrel file is still pending due to confusion over the relationship between react/package.json and react/components/package.json.   I'm thinking one of these can be eliminated.

File: react/package.json [hunk 1]
```diff
@@ -10,10 +10,8 @@
     "install:dependencies": "cd component-demo/showcase && yarn && cd ../storybook && yarn && cd ../../components && yarn",
     "start:showcase": "yarn install:dependencies && yarn link:components && cd component-demo/showcase && yarn start",
     "start:storybook": "yarn install:dependencies && yarn link:components && cd component-demo/storybook && yarn start",
-    "start": "yarn start:showcase",
-    "prettier": "prettier \"components/src/Drawer/**/**.{ts,tsx,js,jsx,json,css,scss,html}\" --write"
+    "start": "yarn start:showcase"
```
Comment (joebochill) [inline react/package.json]: any particular reason for removing the prettier stuff here? Shouldn't it make sense to do the linting and code formatting from the root folder?

Comment (emclaug2) [reply react/package.json]: The prettier stuff can be done at the root level, but the eslint config requires a tsconfig.json to use.  We don't have a global tsconfig, so I had to put the eslint config dependency in 3 different places.   One in showcase, storybook, and components

**Comment by emclaug2** [comment]
I'm marking the FC useCallbacks refactor as an issue on defect so we don't loose track of this.

**Pull request merged**